### PR TITLE
Re-enable knock out reactions

### PR DESCRIPTION
**Pull request by jjkoehorst**
Hello,

Not sure if this is of interest for you but I was playing around with the knockout example and thought it could be useful to allow to re-activate knock-outs. So that if you killed e.coli you can undo the knock-out and revive it. I am no expert in javascript so if I did some obvious cannot do code I am sorry for that...

### CODE REVIEW

File: knockout/knockout.js [hunk 1]
```diff
@@ -153,6 +160,17 @@ function knock_out_reaction(model, reaction_id) {
     throw new Error('Bad reaction ' + reaction_id);
 }
 
+function knock_in_reaction(model, reaction_id) {
+    for (var i = 0, l = model.reactions.length; i < l; i++) {
+        if (model.reactions[i].id == reaction_id) {
+            model.reactions[i].lower_bound = old_model.reactions[i].lower_bound;
+            model.reactions[i].upper_bound = old.model.reactions[i].upper_bound;
```
Comment (zakandrewking) [inline knockout/knockout.js]: looks like a bug here: `old.model` should be `old_model`

**Comment by zakandrewking** [comment]
thanks for your interest in the demos! I noticed a bug. 

After making that fix, the knock_in functionality is not working for me when clicking reaction labels—didn't quite figure out why

As an alternative to adding the functionality here: Have you checked out our new project Escher-FBA?

https://sbrg.github.io/escher-fba

It takes the knockout concept much further and allows both knocking out and resetting reactions

-zak

**Comment by jjkoehorst** [comment]
Thanks for the link! I noticed that when you do maximise / minimise you cannot reset the reaction anymore?

**Comment by zakandrewking** [comment]
closing. if you are interested, please check on the newest version to see if this is still an issue

**Pull request closed without merging**